templates/deployments.yaml

```diff
@@ -982,13 +982,18 @@ spec:
     {{- end }}
 
 {{- range .Values.st2.packs.sensors }}
+  {{- $sensor := omit $.Values.st2sensorcontainer "name" "ref" "postStartScript" }}
+  {{- range $key, $val := . }}
+    {{- $_ := set $sensor $key $val }}
+  {{- end }}
+  {{- $name := print "st2sensorcontainer" (include "hyphenPrefix" $sensor.name) }}
 ---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ $.Release.Name }}-st2sensorcontainer{{ template "hyphenPrefix" .name }}
+  name: {{ $.Release.Name }}-{{ $name }}
   labels:
-    app: st2sensorcontainer{{ template "hyphenPrefix" .name }}
+    app: {{ $name }}
```

Comment:
This chunk was repeated on 5 lines, each of which has to be changed to use `$sensor.name`. So, I consolidated the name definition into one line (above).